<h1>ACTIVITY-1 PALMER PENGUINS DATA ANALYSTICS</h1>|
<hr>

<h4>Analyst:Jayson R. Esmen</h4>

In [157]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings ('ignore')
import matplotlib.pyplot as plt

In [158]:
#libraries

In [159]:
bird = pd.read_csv('datasets\\penguins.csv')

In [160]:
bird

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1>Descriptive Analytics</h1>

In [161]:
bird.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [162]:
bird.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h1>Filtering Column/Data</h1>

In [163]:
bird['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [164]:
Adelie_df = bird[bird['species'] == "Adelie"]

In [165]:
Adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [166]:
Adelie_df = bird['body_mass_g'].mean()

In [167]:
bird['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [168]:
adelie_run_df = bird['species']

<h1>Cleaning the Data Frame</h1>

In [169]:
bird.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h1>What to do with all the NULL VALUES?</h1>

In [170]:
bird['bill_length_mm'].mean()

43.9219298245614

In [171]:
bird['sex'].mode()

0    male
Name: sex, dtype: object

<h1>How to Fillup NULL VALUES</h1>

In [172]:
bird['bill_length_mm']= bird['bill_length_mm'].fillna(bird['bill_length_mm'].mean())
bird['bill_depth_mm']= bird['bill_depth_mm'].fillna(bird['bill_depth_mm'].mean())
bird['flipper_length_mm']= bird['flipper_length_mm'].fillna(bird['flipper_length_mm'].mean())
bird['body_mass_g']= bird['body_mass_g'].fillna(bird['body_mass_g'].mean())

In [173]:
bird['sex'] = bird['sex'].fillna(bird['sex'].mode()[0])

In [174]:
bird.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [175]:
bird['species'] = bird['species'].astype('category')
bird['year'] = bird['year'].astype('category')
bird['island'] = bird['island'].astype('category')
bird['sex'] = bird['sex'].astype('category')

In [176]:
bird.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<h1>EXPLORATORY DATA ANALYSIS</h1>

Ano ang pinaka mabigat ng uri ng penguin

<h1>Which species of Penguin amongst the dataset is the heviest?</h1>

In [177]:
bird['body_mass_g'].mean()

4201.754385964912

In [178]:
bird['species'] == 'Adelie'

0       True
1       True
2       True
3       True
4       True
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

In [179]:
Adelie_df = bird[bird['body_mass_g'] == 'Adelie']['body_mass_g'].mean()

<h2>Insights</h2>
<h4>1.What species has the highest population?</h4>
<h4>2.What is the correlation between bill length and bill depth?</h4>
<h4>3.How many female penguin in the year 2009?</h4>
<h4>4.What species has the heaviest in 2009?</h4>
<h4>5.What is the correlation between body mass and bill length?</h4>
<h4>6.What Adelie species has the highest in the Dream Island?</h4>
<h4>7.What Adelie species has the highest in the TorgersenIsland?</h4>
<h4>8.Which year has the most penguin records?</h4>
<h4>9.Which island has the fewest penguins overalls?</h4>
<h4>10.What is the average mass of Male and Female?</h4>
<h4>11.What is the average bill length of Adelie penguins in Torgersen Island?</h4>
<h4>12.How many penguins have body mass greater than 5000g?</h4>

In [180]:
#1
S_H_P = bird['species'].value_counts()

In [181]:
S_H_P

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

Insights:Therefore the highest population species was Adelie.

In [182]:
#2
C_L_D = bird['bill_length_mm'].corr(bird['bill_depth_mm'])

In [183]:
C_L_D

-0.23505287035553282

Insights:The total correlation between bill length and bill depth was -0.23505287035553282.

In [184]:
#3
F_P_Y = bird[(bird['sex'] == 'female') & (bird['year'] == 2009)].shape[0]

In [185]:
F_P_Y

58

Insights: The total female penguin in the year 2009 is 58.

In [186]:
#4
S_H_Y = bird[bird['year'] == 2009].groupby('species')['body_mass_g'].mean()

In [187]:
S_H_Y

species
Adelie       3664.903846
Chinstrap    3725.000000
Gentoo       5119.358054
Name: body_mass_g, dtype: float64

Insights:The heaviest penguin with the body_mass_g was Gentoo with 5119.358054.

In [188]:
#5
C_B_L = bird['body_mass_g'].corr(bird['bill_length_mm'])

In [189]:
C_B_L

0.5951098244376305

Insights:The correlation between body mass and bill length was 0.5951098244376305.

In [190]:
#6
A_H_D = bird[(bird['species'] == 'Adelie') & (bird['island'] == 'Dream')].shape[0]

In [191]:
A_H_D

56

Insights:The Adelie species has the highest in the Dream Island has a 56 in total.

In [192]:
#7
A_H_T = bird[(bird['species'] == 'Adelie') & (bird['island'] == 'Torgersen')].shape[0]

In [193]:
A_H_T

52

Insights:The Adelie species has the highest in the TorgersenIsland has a 52 in total.

In [194]:
#8
Y_M_P = bird['year'].value_counts()

In [195]:
Y_M_P

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

Insights:The year has the most penguin records was year 2009 with a 120 penguins in total.

In [196]:
#9
A_F_P = bird['island'].value_counts()

In [197]:
A_F_P

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

insights:The island has the fewest penguins overalls is Torgersen with a 52 penguins in total.

In [198]:
#10
A_M_F = bird.groupby('sex')['bill_length_mm'].mean()

In [199]:
A_M_F

sex
female    42.096970
male      45.604156
Name: bill_length_mm, dtype: float64

insights: The average mass of Male and Female, A male with 45.604156 and a female with 42.096970.

In [200]:
#11
A_B_L =bird[(bird['species'] == 'Adelie') & (bird['island'] == 'Torgersen')]['bill_length_mm'].mean()

In [201]:
A_B_L

39.046575573549255

insights:The average bill length of Adelie penguins in Torgersen Island with the bill_length_mm of 39.046575573549255.

In [202]:
#12
B_M_G = bird[bird['body_mass_g'] > 5000].shape[0]

In [203]:
B_M_G

61

insights:The total of penguins have body mass greater than 5000g is 61 penguins in total.